client/components/advanced-filters/index.js

```diff
@@ -0,0 +1,215 @@
+				 */
+			],
+		};
+
+		this.onMatchChange = this.onMatchChange.bind( this );
+		this.onFilterChange = this.onFilterChange.bind( this );
+		this.getSelector = this.getSelector.bind( this );
+		this.getAvailableFilterKeys = this.getAvailableFilterKeys.bind( this );
+		this.addFilter = this.addFilter.bind( this );
+		this.removeFilter = this.removeFilter.bind( this );
+		this.clearAll = this.clearAll.bind( this );
+	}
+
+	onMatchChange( value ) {
+		this.setState( {
+			match: find( this.props.config.matches, match => value === match.value ),
+		} );
+	}
+
+	onFilterChange( key, property, value ) {
+		const activeFilters = this.state.activeFilters.map( activeFilter => {
+			if ( key === activeFilter.key ) {
+				return Object.assign( {}, activeFilter, { [ property ]: value } );
+			}
+			return activeFilter;
+		} );
+
+		this.setState( { activeFilters } );
+	}
+
+	removeFilter( key ) {
+		const activeFilters = [ ...this.state.activeFilters ];
+		const index = findIndex( activeFilters, f => f.key === key );
+		activeFilters.splice( index, 1 );
+		this.setState( { activeFilters } );
+	}
+
+	getTitle() {
+		const { config } = this.props;
+		const { match } = this.state;
+		const label = sprintf( __( 'Orders Match %s Filters', 'wc-admin' ), match.label );
+		return (
+			<Fragment>
+				<span className="screen-reader-text">{ label }</span>
+				<span>{ __( 'Orders Match', 'wc-admin' ) }</span>
+				<SelectControl
+					className="woocommerce-advanced-filters__title-select"
+					options={ config.matches }
+					value={ match.value }
+					onChange={ this.onMatchChange }
+				/>
+				<span>{ __( 'Filters', 'wc-admin' ) }</span>
+			</Fragment>
+		);
+	}
+
+	getSelector( filter ) {
+		const filterConfig = this.props.config.filters[ filter.key ];
+		const { input } = filterConfig;
+		if ( ! input ) {
+			return null;
+		}
+		if ( 'SelectControl' === input.component ) {
+			return (
+				<SelectControl
+					className="woocommerce-advanced-filters__list-select"
+					options={ input.options }
+					value={ filter.value }
+					onChange={ partial( this.onFilterChange, filter.key, 'value' ) }
+				/>
+			);
+		}
+		if ( 'FormTokenField' === input.component ) {
+			return (
+				<FormTokenField
+					value={ filter.value }
+					onChange={ partial( this.onFilterChange, filter.key, 'value' ) }
+				/>
+			);
+		}
+		return null;
+	}
+
+	getAvailableFilterKeys() {
+		const { config } = this.props;
+		const activeFilterKeys = this.state.activeFilters.map( f => f.key );
+		return difference( Object.keys( config.filters ), activeFilterKeys );
+	}
+
+	addFilter( key, onClose ) {
+		const filterConfig = this.props.config.filters[ key ];
+		const newFilter = { key, rule: filterConfig.rules[ 0 ] };
+		if ( filterConfig.input && filterConfig.input.options ) {
+			newFilter.value = filterConfig.input.options[ 0 ];
+		}
+		this.setState( state => {
+			return {
+				activeFilters: [ ...state.activeFilters, newFilter ],
+			};
+		} );
+		onClose();
```

Comment:
Can we also move focus onto the new filter row? VoiceOver leaves you on the +Filter button, and you need to know to tab backwards to get the new filter.